File: 377-combination-sum-iv/377-combination-sum-iv.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int target) {
        vector<unsigned long> v(target+1, 0);
        
        v[0] = 1;
        for(unsigned long i=0; i < target+1; i++)
            for(unsigned long x: nums)
                if (x + i < target+1)
                    v[i+x] += v[i];
        return v[target];
    }
};

File: 377-combination-sum-iv/377-combination-sum-iv.py
class Solution(object):
    def combinationSum4(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        n = [0] * (target+1)
        
        n[0] = 1
        for i in range(target+1):
            for j in nums:
                if i + j < target + 1:
                    n[i + j] += n[i]
        return n[-1]           
        